main/src/cgeo/geocaching/CacheListActivity.java

```diff
@@ -1133,6 +1123,20 @@ private String getDisplayName(final Uri uri) {
     }
 
     public void refreshStored(final List<Geocache> caches) {
+        if (type == CacheListType.OFFLINE && caches.size() > REFRESH_WARNING_THRESHOLD) {
+            Dialogs.confirmYesNo(this, R.string.caches_refresh_all, R.string.caches_refresh_all_warning, new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(final DialogInterface dialog, final int id) {
+                    refreshStoredConfirmed(caches);
+                    dialog.cancel();
```

Comment:
Is this `dialog.cancel()` happening soon enough if another dialog (for cache selection) is created from within `refreshStoredConfirmed()`? Can there be an interaction there?